File: low-level-design/design-patterns/golang/creational/abstract-factory/abstract-factory/azure.go
package abstractfactory

import "fmt"

func NewAzureCompute() *AzureCompute {
	return &AzureCompute{}
}

type AzureCompute struct{}

func (c *AzureCompute) Start() error {
	// Implementation for starting Azure compute resources
	fmt.Println("Starting Azure compute resources")
	return nil
}

func (c *AzureCompute) Stop() error {
	// Implementation for stopping Azure compute resources
	fmt.Println("Stopping Azure compute resources")
	return nil
}

func NewAzureStorage() *AzureStorage {
	return &AzureStorage{}
}

type AzureStorage struct{}

func (s *AzureStorage) Create(bucketName string) error {

	fmt.Println("Creating Azure storage bucket:", bucketName)
	return nil
}

func NewAzureNetwork() *AzureNetwork {

	return &AzureNetwork{}
}

type AzureNetwork struct{}

func (n *AzureNetwork) Configure(name string) error {
	fmt.Println("Configuring Azure network:", name)
	return nil
}

File: low-level-design/design-patterns/python/creational/abstract-factory/azure.py
from resources import Compute,Network,Storage

class AzureCompute(Compute):
    def __init__(self):
        pass
    def Start(self):
        print("Azure Compute instance started")
    def stop(self):
        print("Azure Compute instance stopped")

class AzureStorage(Storage):
    def __init__(self):
        pass
    def Create(self,bucketName:str):
        print(f"Azure Storage bucket {bucketName} created")

class AzureNetwork(Network):
    def __init__(self):
        pass
    def Configure(self,subnetName:str):
        print(f"Azure Network subnet {subnetName} configured")
